packages/contracts-bedrock/scripts/Deployer.sol

```diff
@@ -561,12 +564,23 @@ abstract contract Deployer is Script {
     }
 
     /// @dev Returns the value of the internal `_initialized` storage slot for a given contract.
-    function loadInitializedSlot(string memory _contractName, bool _isProxy) public returns (uint8 initialized_) {
-        StorageSlot memory slot = getInitializedSlot(_contractName);
-        if (_isProxy) {
-            _contractName = string.concat(_contractName, "Proxy");
+    function loadInitializedSlot(string memory _contractName) public returns (uint8 initialized_) {
+        address contractAddress;
+        // Check if the contract name ends with `Proxy` and, if so, get the implementation address
+        if (LibString.endsWith(_contractName, "Proxy")) {
+            contractAddress = EIP1967Helper.getImplementation(getAddress(_contractName));
+            _contractName = LibString.slice(_contractName, 0, bytes(_contractName).length - 5);
+            // If the EIP1967 implementation address is 0, we try to get the implementation address from legacy
+            // AddressManager, which would work if the proxy is ResolvedDelegateProxy like L1CrossDomainMessengerProxy.
+            if (contractAddress == address(0)) {
+                contractAddress =
+                    AddressManager(mustGetAddress("AddressManager")).getAddress(LibString.concat("OVM_", _contractName));
```

Comment:
nit, can use native solidity here
```suggestion
                    AddressManager(mustGetAddress("AddressManager")).getAddress(string.concat("OVM_", _contractName));
```

Reply:
this was originally the case but switched to solady's due to gas efficiency, given that the library is a dep now. should we switch back to native solidity here?

Reply:
IMO we should, since this is a script we don't care about gas efficiency, so better to keep code simple and minimize deps when there's native solidity support. But this is definitely a nit

Reply:
I agree -- will update. thanks!